internals/notifier.py

```diff
@@ -254,6 +254,104 @@ def get_feature_starrers(self, feature_id):
     return user_prefs
 
 
+class FeatureAccuracyHandler(basehandlers.FlaskHandler):
+  SEND_NOTIFICATIONS = False
+
+  CHROME_RELEASE_SCHEDULE_URL = (
+'https://chromiumdash.appspot.com/fetch_milestone_schedule')
+  ACCURACY_AS_OF_WEEKS = 4
+  MILESTONE_FIELDS = (
+    'dt_milestone_android_start',
+    'dt_milestone_desktop_start',
+    'dt_milestone_ios_start',
+    'dt_milestone_webview_start',
+    'ot_milestone_android_start',
+    'ot_milestone_desktop_start',
+    'ot_milestone_webview_start',
+    'shipped_android_milestone',
+    'shipped_ios_milestone',
+    'shipped_milestone',
+    'shipped_webview_milestone'
+  )
+  TEMPLATE_PATH = 'accuracy_notice_email.html'
+   
+  def get_template_data(self):
+    """Sends notifications to users requesting feature updates for accuracy."""
+
+    # Do not send notifications until client-side changes are made.
+    # TODO(danielrsmith): This check should be removed when the new page to
+    # verify data accuracy is implemented.
+    if not self.SEND_NOTIFICATIONS:
+      return {'message': '0 emails sent or logged.'}
+
+    features_to_notify = self._determine_features_to_notify()
+    email_tasks = self._build_email_tasks(features_to_notify)
+    send_emails(email_tasks)
+    return {'message': f'{len(email_tasks)} emails sent or logged.'}
+    
+  def _determine_features_to_notify(self):
+    # 'current' milestone is the next stable milestone that hasn't landed.
+    # We send notifications to any feature planned for beta or stable launch
+    # in the next 8 weeks. Beta for (current + 2) starts in roughly 8 weeks.
+    # So we check if any features have launches in these 3 milestones
+    # (current, current + 1, and current + 2).
+    try:
+      resp = requests.get(f'{self.CHROME_RELEASE_SCHEDULE_URL}?mstone=current')
+    except requests.RequestException as e:
+      raise e
+    mstone_info = json.loads(resp.text)
+    mstone = int(mstone_info['mstones'][0]['mstone'])
+    
+    features = models.Feature.query(models.Feature.deleted == False).fetch(None)
+    features_to_notify = []
+    now = datetime.now()
+    accuracy_as_of_delta = timedelta(weeks=self.ACCURACY_AS_OF_WEEKS)
+    for feature in features:
+      accuracy_as_of_date = None
+      # Format accuracy date to compare dates.
+      if feature.accurate_as_of is not None:
+        accuracy_as_of_date = datetime.strptime(
```

Comment:
Why do we need to convert to a string and then parse the data back into date time object?  Could we use feature.accurate_as_of directly in the if-statement below?

Reply:
This was an old code block when I wasn't certain if the `ndb.DateTimeProperty` could be used as a true datetime object. I forgot to refactor it - thanks for pointing it out!